honeybee_energy/material/opaque.py

```diff
@@ -330,6 +336,7 @@ def __copy__(self):
             self.specific_heat, self.roughness, self.thermal_absorptance,
             self.solar_absorptance, self._visible_absorptance)
         new_material._display_name = self._display_name
+        new_material._user_data = None if self.user_data is None else self.user_data.copy()
```

Comment:
You will need something like this in the `__copy__` functions of all materials and constructions with user data. Otherwise it will get lost as the object is passed from one Grasshopper component to another.